clean/ca/los_angeles_sheriff.py

```diff
@@ -0,0 +1,227 @@
+import logging
+import time
+from pathlib import Path
+
+import requests
+
+from .. import utils
+from ..cache import Cache
+from .config.los_angeles_sheriff import (
+    detail_payload,
+    detail_request_headers,
+    index_payload,
+    index_request_headers,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class Site:
+    name = "Los Angeles Sheriff's Department"
+
+    def __init__(self, data_dir=utils.CLEAN_DATA_DIR, cache_dir=utils.CLEAN_CACHE_DIR):
+        self.siteslug = "ca_los_angeles_sheriff"
+        self.rooturl = "https://lasdsb1421.powerappsportals.us"
+        self.filestoignore = [
+            "index",
+            "timestamplog",
+            self.siteslug,
+            "caseindex",
+        ]  # What cached JSON files aren't page-level JSONs?
+        self.base_url = "https://lasd.org/"
+        self.disclosure_url = "https://lasdsb1421.powerappsportals.us/"
+        self.data_dir = data_dir
+        self.cache_dir = cache_dir
+        self.cache = Cache(cache_dir)
+        self.subpages_dir = cache_dir / (self.siteslug + "/subpages")
+        for localdir in [self.cache_dir, self.data_dir, self.subpages_dir]:
+            utils.create_directory(localdir)
+
+    def scrape_meta(self, throttle: int = 0) -> Path:
+        rawindex = self._fetch_index()
+        oldtimestamps = self._fetch_old_timestamps()
+        indextimes = self._build_timestamps(rawindex)
+        detailtodo = self._build_detail_todo(indextimes, oldtimestamps)
+        self._fetch_detail_pages(detailtodo, throttle)
+        self._save_timestamps(indextimes)
+        caseindex = self._build_caseindex(rawindex)
+        assetlist = self._build_assetlist(caseindex)
+        assetlist_filename = self._save_assetlist(assetlist)
+        return assetlist_filename
+
+    def _fetch_index(self):
+        indexjsonurl = "https://lasdsb1421.powerappsportals.us/_services/entity-grid-data.json/f46b70cc-580b-4f1a-87c3-41deb48eb90d"
+        r = requests.post(
+            indexjsonurl,
+            headers=index_request_headers,
+            data=index_payload,
+        )
+        targetfilename = f"{self.siteslug}/index.json"
+        self.cache.write_binary(targetfilename, r.content)
+        # HEY!
```

Comment:
there's a common convention for this! will merge this to expedite
```suggestion
        # FIXME:
```